src/main/java/net/mcreator/ui/procedure/TextProcedureSelector.java

```diff
@@ -0,0 +1,278 @@
+		} else {
+			if (helpContext == null)
+				top.add("North", nameLabel);
+			else
+				top.add("North", HelpUtils.wrapWithHelpButton(helpContext, nameLabel, SwingConstants.LEFT));
+		}
+
+		top.add("South", depslab);
+
+		add.setContentAreaFilled(false);
+		add.setOpaque(false);
+		add.setMargin(new Insets(0, 0, 0, 0));
+		add.addActionListener(e -> {
+			String procedureNameString = "";
+			if (mcreator.mcreatorTabs.getCurrentTab().getContent() instanceof ModElementGUI) {
+				StringBuilder procedureName = new StringBuilder(
+						((ModElementGUI<?>) mcreator.mcreatorTabs.getCurrentTab().getContent()).getModElement()
+								.getName());
+				String[] parts = eventName.replaceAll("\\(.*\\)", "").split(" ");
+				for (String part : parts) {
+					procedureName.append(StringUtils.uppercaseFirstLetter(part));
+				}
+				procedureNameString = JavaConventions.convertToValidClassName(
+						procedureName.toString().replace("When", ""));
+			}
+
+			procedureNameString = VOptionPane.showInputDialog(mcreator, L10N.t("action.procedure.enter_procedure_name"),
+					L10N.t("action.procedure.new_procedure_dialog_title"), null, new OptionPaneValidatior() {
+						@Override public ValidationResult validate(JComponent component) {
+							return UniqueNameValidator.createModElementNameValidator(mcreator.getWorkspace(),
+									(VTextField) component, L10N.t("common.mod_element_name")).validate();
+						}
+					}, L10N.t("action.procedure.create_procedure"), UIManager.getString("OptionPane.cancelButtonText"),
+					procedureNameString);
+
+			if (procedureNameString != null) {
+				ModElement element = new ModElement(mcreator.getWorkspace(), procedureNameString,
+						ModElementType.PROCEDURE);
+				ModElementGUI<?> newGUI = ModElementType.PROCEDURE.getModElementGUI(mcreator, element, false);
+				if (newGUI != null) {
+					newGUI.showView();
+					newGUI.setModElementCreatedListener(generatableElement -> {
+						String modName = JavaConventions.convertToValidClassName(
+								generatableElement.getModElement().getName());
+						refreshList();
+						setSelectedProcedure(modName);
+					});
+					mcreator.getApplication().getAnalytics().async(() -> mcreator.getApplication().getAnalytics()
+							.trackEvent(AnalyticsConstants.EVENT_NEW_MOD_ELEMENT,
+									ModElementType.PROCEDURE.getReadableName(), null, null));
+				}
+			}
+		});
+
+		edit.setMargin(new Insets(0, 0, 0, 0));
+		edit.setOpaque(false);
+		edit.setContentAreaFilled(false);
+		edit.addActionListener(e -> {
+			if (getSelectedProcedure() != null) {
+				ModElement selectedProcedureAsModElement = mcreator.getWorkspace()
+						.getModElementByName(getSelectedProcedure().getName());
+				ModElementGUI<?> modeditor = selectedProcedureAsModElement.getType()
+						.getModElementGUI(mcreator, selectedProcedureAsModElement, true);
+				if (modeditor != null)
+					modeditor.showView();
+			}
+		});
+
+		if (fixedText != null)
+			add("East", PanelUtils.totalCenterInPanel(
```

Comment:
I think you should put the fixed text input on another line (South). It makes the UI much bigger and stretched. 
![image](https://user-images.githubusercontent.com/38427877/200134184-7e7a430a-7533-4f7a-ac28-2aa35169799b.png)